ecs-init/cache/cache.go

```diff
@@ -222,53 +240,39 @@ func (d *Downloader) DownloadAgent() error {
 }
 
 func (d *Downloader) getPublishedMd5Sum() (string, error) {
-	tempFile, err := d.fs.TempFile(config.CacheDirectory(), "ecs-agent.tar.md5")
+	tempMd5FileName, err := d.s3Downloader.downloadFile(config.AgentRemoteTarballMD5Key())
 	if err != nil {
+		log.Debugf("Failed to download md5 file for published tarball: %v", err)
 		return "", err
 	}
-	log.Debugf("Created temporary file for md5sum: %s", tempFile.Name())
-	defer func() {
-		log.Debugf("Removing temp file %s", tempFile.Name())
-		d.fs.Remove(tempFile.Name())
-	}()
-	defer tempFile.Close()
 
-	_, err = d.s3downloader.Download(tempFile, &s3.GetObjectInput{
-		Bucket: aws.String(config.AgentRemoteBucketName),
-		Key:    aws.String(config.AgentRemoteTarballMD5Key()),
-	})
+	tempMd5File, err := d.fs.Open(tempMd5FileName)
 	if err != nil {
+		log.Debugf("Failed to open temporary md5 file: %v", err)
 		return "", err
 	}
+	defer func() { // clean up temp file
+		log.Debugf("Removing temp file %s", tempMd5FileName)
+		d.fs.Remove(tempMd5FileName)
+	}()
 
-	body, err := d.fs.ReadAll(tempFile)
+	body, err := d.fs.ReadAll(tempMd5File)
 	if err != nil {
+		log.Debugf("Failed to read from temporary md5 file: %v", err)
```

Comment:
Same as above. Can you please wrap the original error with something like `errors.Wrapf(err, "failed to read from temporary md5 file")`?